### PR TITLE
Tests Merge - Accounts & Nexus

### PR DESCRIPTION
**Pull request by syahirAmali**

### CODE REVIEW

**Review by Cosmodude — COMMENTED** [review]
Good, I will merge after the audit, please keep working in test branch

File: contracts/ERC6551Account.sol [hunk 1]
```diff
@@ -184,11 +183,14 @@ contract ReferralHandlerERC6551Account is
 
     // @audit - Users can create multiple accounts, and refer each other with no limits, and update the tiers of their referral trees
     // @audit - this enables these accounts to be tiered up without any real referrals
-    function updateReferralTree(uint8 refDepth, uint8 _tier) external {
+    // CIRTICAL - NOTE: Change param, check tiers, call the account, and get the actual tier from the contract
+    function updateReferralTree(uint8 refDepth, address referralHandler) external {
```
[the remark this answers is not in the document]
Comment (syahirAmali) [reply contracts/ERC6551Account.sol:187]: If we do that other contracts wont be able to update on behalf of accounts, there's no where that does this yet. If you there's not going to be a scenario for that, I'll make this change

[the remark this answers is not in the document]
Comment (Cosmodude) [reply contracts/ERC6551Account.sol:187]: Yeah, only the account itself, should be able to update referrers above

**Review by Cosmodude — COMMENTED** [review]
If the quest is 1000 AVAX, the seeker should deposit 103% (1030 AVAX);
30 AVAX go to tax

File: contracts/interfaces/ITaxManager.sol [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
 
 interface ITaxManager {
+    struct SeekerFees {
+        uint256 referralRewards;
+        uint256 platformRevenue;
+    }
+
+    function getSeekerFees() external view returns (SeekerFees memory);
+
+    function getReferralTaxReceiver() external view returns (address);
```
Comment (Cosmodude) [inline contracts/interfaces/ITaxManager.sol:12]: referral tax goes to referrers and leftovers (40 percent or more) go to platform pool, different with the below one though

File: contracts/TaxManager.sol [hunk 3]
```diff
@@ -15,24 +15,19 @@ contract TaxManager is ITaxManager {
...
-    address public devPool;
-    address public rewardAllocationPool;
-    address public perpetualPool;
-    address public tierPool;
     address public revenuePool;
-    address public marketingPool;
 
-    uint256 public seekerTaxRate;
+    // Tax Receiver Addresses
+    address private referralTaxReceiver;
```
Comment (Cosmodude) [inline contracts/TaxManager.sol:22]: good logic for pools
initial platform tax goes to `platformTaxReceiver`

leftovers from referral tax go to `referralTaxReceiver`

File: contracts/TaxManager.sol [hunk 4]
```diff
@@ -15,24 +15,19 @@ contract TaxManager is ITaxManager {
 
     address public custodian;
 
-    address public seekerTaxPool;
     address public solverTaxPool;
-    address public maintenancePool;
-    address public devPool;
-    address public rewardAllocationPool;
-    address public perpetualPool;
```
Comment (Cosmodude) [inline contracts/TaxManager.sol:23]: Thanks for cleaning this up

File: contracts/TaxManager.sol [hunk 5]
```diff
@@ -84,145 +84,98 @@ contract TaxManager is ITaxManager {
...
         uint256 first,
         uint256 second,
         uint256 third,
         uint256 fourth
-    ) external onlyCustodian {
+    ) 
+        external 
+        onlyCustodian 
+        validTaxRate(first)
+        validTaxRate(second)
```
Comment (Cosmodude) [inline contracts/TaxManager.sol:169]: great sanity check

**Review by Cosmodude — APPROVED** [review]
Please, pay attention to this comment:
**If the quest is 1000 AVAX, the seeker should deposit 103% (1030 AVAX);
30 AVAX go to tax**

https://www.figma.com/file/z8CjpkZRh8TxXb69f4T5iK/The-Guild-Fees-%26-Taxes-%26-Rewards?type=whiteboard&node-id=0-1&t=CUHgqYU4ebkMpgpE-0

File: contracts/TaxManager.sol (hunk 3 above)
Comment (Cosmodude) [reply contracts/TaxManager.sol:22]:
```
// Tax Receiver Addresses
    address private referralTaxReceiver;
    address private platformTaxReceiver;
```

Why are the addresses private btw?

**Pull request merged**